src/app/util/attribute-storage.cpp

```diff
@@ -261,9 +261,43 @@ uint16_t emberAfGetDynamicIndexFromEndpoint(EndpointId id)
     return kEmberInvalidEndpointIndex;
 }
 
+const EmberAfCluster * getClusterTypeDefinition(EndpointId endpointId, ClusterId clusterId, EmberAfClusterMask mask)
+{
+    uint16_t index = emberAfIndexFromEndpointIncludingDisabledEndpoints(endpointId);
+    if (index != kEmberInvalidEndpointIndex)
+    {
+        return emberAfFindClusterInType(emAfEndpoints[index].endpointType, clusterId, mask, nullptr);
+    }
+    // not found
+    return nullptr;
+}
+
+CHIP_ERROR setupDynamicEndpointDeclaration(EmberAfEndpointType & endpointType, EndpointId templateEndpointId,
+                                           const Span<const ClusterId> & templateClusterIds)
+{
+    // allocate cluster list
+    endpointType.clusterCount = static_cast<uint8_t>(templateClusterIds.size());
+    endpointType.cluster      = new EmberAfCluster[endpointType.clusterCount];
```

Comment:
where is the corresponding `delete[]` for this new? Please add a comment about lifetime and ownership of these.

Reply:
Also wondering if heap or PlatformNew is the correct path here. Ideally there would be a way to have some std::unique_ptr for auto memory management if we find a way to do that.

Reply:
There was no matching delete[]. Now I added `emberAfResetDynamicEndpointDeclaration` as a proper counterpart to `emberAfSetupDynamicEndpointDeclaration`.

I omitted that initially, because I see no real use case, and the implications of actually getting rid of a dynamic endpoint in the rest of the code seemed so far reaching that it did not seem worthwhile to consider at all.